### ISSUE TITLE
Add support for audio recording

### ISSUE THREAD

**Issue by kapoorlakshya** [kapoorlakshya/screen-recorder#15]
See if audio recording can be supported through `advanced`. If not, look into implementing parameters for it.

**Comment by kapoorlakshya** [comment]
macOS - https://trac.ffmpeg.org/wiki/Capture/Desktop

Linux - https://trac.ffmpeg.org/wiki/Capture/ALSA

Windows - https://medium.com/@colten_jackson/simple-free-screen-capture-with-ffmpeg-3aeddbeb161b

**Comment by sh3p42d** [comment]
Hello! Thanks for the gem!
Is it possible to record video with audio now? I added settings to 'advanced' but got an error 'Unable to find a suitable output format for ***'. If its possible, I dont undestand where and what settings need to be set to do this. I use linux, if its important.

**Comment by kapoorlakshya** [comment]
Hi @sh3p42d, can you share the advanced configs (code for it) that you have tried? There is an example showing the audio configs on Linux over here: https://trac.ffmpeg.org/wiki/Capture/Desktop

> If you need audio too, you can use ALSA (see Capture/ALSA for more info):
>```ffmpeg -video_size 1024x768 -framerate 25 -f x11grab -i :0.0+100,200 -f alsa -ac 2 -i hw:0 output.mkv```

I'll look into updating the gem with this feature very soon.

**Comment by sh3p42d** [comment]
Thank you for efficiency.
I want to run this command "ffmpeg -video_size 1024x768 -framerate 25 -f x11grab -i :0.0+100,200 -f pulse -ac 2 -i default output.mp4"

This is my configs:
```
advanced = { 
    input: {
        video_size: '1024x768',
        framerate: '25',
        ac: '2'
    }
}
@recorder = ScreenRecorder::Desktop.new(input: ':0.0+100,200 pulse default', output: 'output.mp4', advanced: advanced)
```

How can i set the advanced configs for '-f pulse -ac 2 -i default'? Every time I got 'Unable to find a suitable output format for'.

**Comment by kapoorlakshya** [comment]
@sh3p42d I just pushed some code which will allow you to record audio with the following configs:

```ruby
advanced = { 'f': 'pulse', 'ac': 2, 'i': 'default' } # Uses default sound output device
@recorder = ScreenRecorder::Desktop.new(output: 'output.mp4', advanced: advanced)
```

You can test it in your current project by adding this to your `Gemfile`:

```
gem 'screen-recorder', git: 'https://github.com/kapoorlakshya/screen-recorder.git', ref: 'correctly_parse_advanced_settings'
```

And you don't need to provide an `input` value if you're not using multiple screens. The default is set to `:0` which should work on most systems. You can check your input number by executing `echo $DISPLAY` in the terminal.

I'll implement a better solution to record audio when I have more free time. Let me know if you run into any problems.

**Comment by kapoorlakshya** [comment]
I have pushed some more code to properly support and document the configs for capturing audio: https://github.com/kapoorlakshya/screen-recorder/compare/15_capture_audio

Will merge it after testing it on Windows and macOS over the weekend.

**Comment by kapoorlakshya** [comment]
Closed by 1ba5e556bcb0b90f6525cdc845667543c02c79ee.
